circuits/move/move.func.circom

```diff
@@ -110,34 +110,42 @@ template CheckRsrc(N_TL_ATRS, RSRC_IDX, PUBX_IDX, PUBY_IDX, UNOWNED, SYS_BITS) {
     signal ontoSelfOrUnowned <== OR()(ontoSelf, ontoUnowned);
     signal ontoEnemy <== NOT()(ontoSelfOrUnowned);
     signal ontoMoreOrEq <== GreaterEqThan(SYS_BITS)([tTo[RSRC_IDX], 
-        uFrom[RSRC_IDX] - tFrom[RSRC_IDX]]);
+        tFrom[RSRC_IDX] - uFrom[RSRC_IDX]]);
     signal ontoLess <== NOT()(ontoMoreOrEq);
 
+    // From tile must stay player's, player cannot be UNOWNED
+    signal fromOwnership <== IsEqual()([tFromPub, uFromPub]);
+    signal fromOwnershipWrong <== NOT()(fromOwnership);
+    signal playerUnowned <== IsEqual()([tFromPub, UNOWNED]);
```

Comment:
i see- people know the private key of {x: 0, y: 0}

maybe we set unowned to the public key {x: H('network'), y: H('states')} so we can avoid this check? it's cleaner so we don't have to worry about people claiming UNOWNED tiles ever, even in future updates

Reply:
MACI does this w/ https://github.com/privacy-scaling-explorations/maci/blob/78609349aecd94186216ac8743d61b1cb81a097f/crypto/README.md?plain=1#L11C1-L11C1. Seems like best practice for ensuring unowned leaves are really unowned